src/pages/workspace/reimburse/WorkspaceRateAndUnitPage.js

```diff
@@ -33,6 +40,45 @@ class WorkspaceRateAndUnitPage extends React.Component {
         super(props);
         this.submit = this.submit.bind(this);
         this.validate = this.validate.bind(this);
+
+        this.state = {
+            rate: 0,
+            unit: 'mi',
+        };
+    }
+
+    componentDidMount() {
+        const distanceCustomUnit = _.find(lodashGet(this.props, 'policy.customUnits', {}), (unit) => unit.name === CONST.CUSTOM_UNITS.NAME_DISTANCE);
```

Comment:
seems you duplicate line 51 - 57 inside componentDidUpdate
let's create a separate function and put all code inside it, and re-use in both didMount and didUpdate

Reply:
Extracted this logic into separate function